### PR TITLE
Handle SIGINT from parent to terminate child process

### LINKED ISSUE #1487: Close children processes gracefully upon SIGINT (and other termination signals)
I don't know if it is expected, but we get:
```
t=2023-12-07T19:34:31+0100 lvl=eror msg="failed to close pre-image server" err="signal: interrupt"
```
when stopping the zkVM with `CTRL-C`.

Not urgent IMHO, and definitely not in the scope of this PR, but would be nice to open a follow-up.

_Originally posted by @dannywillems in https://github.com/o1-labs/proof-systems/issues/1469#issuecomment-1845941574_

### PR DESCRIPTION
**Pull request by rbonichon**
Should fix #1487

### CODE REVIEW

**Comment by dannywillems** [comment]
Is it still required? @rbonichon

**Comment by rbonichon** [comment]
> Is it still required? @rbonichon

Guess so for it's still not handled.

File: optimism/src/main.rs [hunk 1]
```diff
@@ -77,6 +85,10 @@ pub fn main() -> ExitCode {
 
     let mut env = mips_witness::Env::<ark_bn254::Fr>::create(cannon::PAGE_SIZE as usize, state, po);
 
+    // Install signal catcher
+    let term = Arc::new(AtomicBool::new(false));
+    signal_hook::flag::register(signal_hook::consts::SIGTERM, Arc::clone(&term)).unwrap();
```
Comment (volhovm) [inline optimism/src/main.rs:90]: This seems exactly equivalent to the code on lines 68-70, I assume that's not on purpose?

File: optimism/src/main.rs [hunk 2]
```diff
@@ -124,7 +136,7 @@ pub fn main() -> ExitCode {
             next: std::array::from_fn(|_| Vec::with_capacity(domain_size)),
         };
 
-    while !env.halt {
+    while !term.load(Ordering::Relaxed) && !env.halt {
```
Comment (volhovm) [inline optimism/src/main.rs:139]: I'm not exactly sure I understand what's happening... I assume after registering the hook, SIGINT will not work as usual, but will instead just modify the mutable variable? 
- In this case: Is it intended that the Keccak proof generation (lines 256-270) do not depend on `term`? Will that part be interrupted by SIGINT?

**Review by volhovm — COMMENTED** [review]
Could you please clarify the logic? I'm unsure whether if all the parts of `main` are properly interrupted by SIGINT.

**Comment by dannywillems** [comment]
Closing as old and stale.

**Pull request closed without merging**